src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/SubscriptionRouter.cs

```diff
@@ -15,19 +14,11 @@ public SubscriptionRouter(Publishers publishers, EndpointInstances endpointInsta
             this.transportAddressTranslation = transportAddressTranslation;
         }
 
-        public async Task<IEnumerable<string>> GetAddressesForEventType(Type messageType)
+        public IEnumerable<string> GetAddressesForEventType(Type messageType)
         {
-            var results = new HashSet<string>();
-            foreach (var publisherAddress in publishers.GetPublisherFor(messageType))
-            {
-                var addresses = await publisherAddress.Resolve(endpoint => endpointInstances.FindInstances(endpoint), i => transportAddressTranslation(i)).ConfigureAwait(false);
-                foreach (var address in addresses)
-                {
-                    results.Add(address);
-                }
-            }
-
-            return results;
+            var publisher = publishers.GetPublisherFor(messageType);
```

Comment:
Can you help me translate what's described in #3911 to an acceptance test? My change did not break any tests so I am afraid we might be missing something.